presto-main/src/test/java/io/prestosql/sql/planner/TestLogicalPlanner.java

```diff
@@ -255,6 +258,22 @@ public void testJoinWithOrderBySameKey()
                                         tableScan("lineitem", ImmutableMap.of("LINEITEM_OK", "orderkey"))))));
     }
 
+    @Test
+    public void testTopNPushdownToJoinSource()
+    {
+        assertPlan("SELECT n.name, r.name FROM nation n LEFT JOIN region r ON n.regionkey = r.regionkey ORDER BY n.comment LIMIT 1",
+                anyTree(
+                        project(
+                                topN(1, ImmutableList.of(sort("N_COMM", ASCENDING, LAST)), TopNNode.Step.FINAL,
+                                        anyTree(
+                                                join(LEFT, ImmutableList.of(equiJoinClause("N_KEY", "R_KEY")),
+                                                        project(
+                                                                topN(1, ImmutableList.of(sort("N_COMM", ASCENDING, LAST)), TopNNode.Step.PARTIAL,
+                                                                        tableScan("nation", ImmutableMap.of("N_NAME", "name", "N_KEY", "regionkey", "N_COMM", "comment")))),
+                                                        anyTree(
```

Comment:
Nits:

replace with `anyNot(TopNode.class, anyTree(...))`?
The best would be `anyTreeNot(TopNode.class, tableScan(...)`

Reply:
can't find `anyTreeNot()`

Reply:
Because it does not exists yet. It would be the best if it would have existed.

Reply:
I think it could be added in another PR. I could give it a try later.